documentation/seed-node.md

```diff
@@ -0,0 +1,255 @@
+`syncPublishScreenAndServe` commands. 
+
+## Setup
+
+These commands work in conjunction with a [state-diffing full Geth node](https://github.com/vulcanize/go-ethereum/tree/rpc_statediffing)
+and IPFS.
+
+### IPFS
+To start, download and install [IPFS](https://github.com/vulcanize/go-ipfs)
+
+`go get github.com/ipfs/go-ipfs`
+
+`cd $GOPATH/src/github.com/ipfs/go-ipfs`
+
+`make install`
+
+If we want to use Postgres as our backing datastore, we need to use the vulcanize fork of go-ipfs. This fork supports
+the Postgres datastore plugin and has been adjusted to use `dep` instead of Go modules since Go modules cannot work with
+un-versioned forks and we need to use an un-versioned fork of go-ipfs-config.
+
+Start by adding the fork and switching over to it:
+
+`git remote add vulcanize https://github.com/vulcanize/go-ipfs.git`
+
+`git fetch vulcanize`
+
+`git checkout -b postgres_update vulcanize/postgres_update`
+
+Now install this fork of ipfs, first be sure to remove any previous installation. It is important to use the below command
+instead of using `make install`, as `make install` will default to using Go modules and this will wreck our dependencies
+since Go modules cannot work with un-versioned forks.
+
+`go install ./cmd/ipfs`
+
+Check that is installed properly by running
+
+`ipfs`
+
+You should see the CLI info/help output.
+
+And now we initialize with the `postgresds` profile.
+If ipfs was previously initialized we will need to remove the old profile first.
+We also need to provide env variables for the postgres connection: 
+
+We can either set these manually, e.g.
+```bash
+export IPFS_PGHOST=
+export IPFS_PGUSER=
+export IPFS_PGDATABASE=
+export IPFS_PGPORT=
+export IPFS_PGPASSWORD=
+```
+
+And then run the ipfs command
+
+`ipfs init --profile=postgresds`
+
+Or we can use the pre-made script at `GOPATH/src/github.com/ipfs/go-ipfs/misc/utility/ipfs_postgres.sh`
+which has usage: 
+
+`./ipfs_postgres.sh <IPFS_PGHOST> <IPFS_PGPORT> <IPFS_PGUSER> <IPFS_PGDATABASE>"`
+
+and will ask us to enter the password, avoiding storing it to an ENV variable.
+
+Once we have initialized ipfs, that is all we need to do with it- we do not need to run a daemon during the subsequent processes.
+
+### Geth 
+For Geth, we currently *require* a special fork, and we can set this up as follows:
+
+Begin by downloading geth and switching to the vulcanize/rpc_statediffing branch
+
+`go get github.com/ethereum/go-ethereum`
+
+`cd $GOPATH/src/github.com/ethereum/go-ethereum`
+
+`git remote add vulcanize https://github.com/vulcanize/go-ethereum.git`
+
+`git fetch vulcanize`
+
+`git checkout -b rpc_statediffing vulcanize/rpc_statediffing`
+
+Now, install this fork of geth (make sure any old versions have been uninstalled/binaries removed first)
+
+`make geth`
+
+And run the output binary with statediffing turned on:
+
+`cd $GOPATH/src/github.com/ethereum/go-ethereum/build/bin`
+
+`./geth --statediff --statediff.streamblock --ws --syncmode=full`
```

Comment:
Just out of curiosity, is `--statediff.streamblock` required for the seed node because we the full block data instead of just the state diff, right?

Reply:
Yeah that is correct!

Reply:
@elizabethengelman What does the `streamblock` flag do?

Reply:
It configures the statediff subscription to send along headers, transactions, and receipts in addition to the state and storage diffs.